src/Compilers/CSharp/Portable/Binder/Binder_TupleOperators.cs

```diff
@@ -0,0 +1,386 @@
+            {
+                boolConversion = best.Conversion;
+                boolOperator = best.Signature;
+                return;
+            }
+
+            // It did not. Give a "not convertible to bool" error.
+
+            GenerateImplicitConversionError(diagnostics, node, conversion, comparisonResult, boolean);
+            boolConversion = Conversion.NoConversion;
+            boolOperator = default;
+            return;
+        }
+
+        private TupleBinaryOperatorInfo BindTupleDynamicBinaryOperatorSingleInfo(BinaryExpressionSyntax node, BinaryOperatorKind kind,
+            BoundExpression left, BoundExpression right, DiagnosticBag diagnostics)
+        {
+            // This method binds binary == and != operators where one or both of the operands are dynamic.
+            Debug.Assert((object)left.Type != null && left.Type.IsDynamic() || (object)right.Type != null && right.Type.IsDynamic());
+
+            bool hasError = false;
+            bool leftValidOperand = IsLegalDynamicOperand(left);
+            bool rightValidOperand = IsLegalDynamicOperand(right);
+
+            if (!leftValidOperand || !rightValidOperand)
+            {
+                // Operator '{0}' cannot be applied to operands of type '{1}' and '{2}'
+                Error(diagnostics, ErrorCode.ERR_BadBinaryOps, node, node.OperatorToken.Text, left.Display, right.Display);
+                hasError = true;
+            }
+
+            BinaryOperatorKind elementOperatorKind = hasError ? kind : kind.WithType(BinaryOperatorKind.Dynamic);
+            TypeSymbol dynamicType = Compilation.DynamicType;
+
+            // We'll end up dynamically invoking operators true and false, so no result conversion. We'll deal with that during lowering.
+            return new TupleBinaryOperatorInfo.Single(dynamicType, dynamicType, elementOperatorKind,
+                methodSymbolOpt: null, boolConversion: Conversion.Identity, boolOperator: default);
+        }
+
+        private TupleBinaryOperatorInfo BindTupleBinaryOperatorNestedInfo(BinaryExpressionSyntax node, BinaryOperatorKind kind,
+            ImmutableArray<BoundExpression> left, ImmutableArray<BoundExpression> right,
+            bool nullable, DiagnosticBag diagnostics)
+        {
+            int length = left.Length;
+            Debug.Assert(length == right.Length);
+
+            var operatorsBuilder = ArrayBuilder<TupleBinaryOperatorInfo>.GetInstance(length);
+
+            for (int i = 0; i < length; i++)
+            {
+                operatorsBuilder.Add(BindTupleBinaryOperatorInfo(node, kind, left[i], right[i], diagnostics));
+            }
+
+            var compilation = this.Compilation;
+            TypeSymbol leftTupleType = MakeConvertedType(operatorsBuilder, node.Left, left, nullable, compilation, diagnostics, isRight: false);
+            TypeSymbol rightTupleType = MakeConvertedType(operatorsBuilder, node.Right, right, nullable, compilation, diagnostics, isRight: true);
```

Comment:
Consider passing an `ImmutableArray<TupleBinaryOperatorInfo>` for the operators, for consistency with the `elements` array. (We're creating the `ImmutableArray` below currently.)